src/guides/v2.3/extension-dev-guide/framework/math-random.md

```diff
@@ -0,0 +1,44 @@
+---
+group: php-developer-guide
+title: Math Random
+contributor_name: Adarsh Manickam
+contributor_link: https://github.com/drpayyne
+---
+
+## Overview
+
+The [`Magento\Framework\Math\Random`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Math/Random.php){:target="\_blank"} library provides the ability to generate random values and hashes.
+
+## Usage
+
+| Method            | Description                                                                                                                                                                          |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `getRandomString` | Generates a random string with a given string length and an optional character pool. Without the optional character pool, it uses both cases of the English alphabet and all digits. |
+| `getRandomNumber` | Generates a random number with an optional range. The range defaults to 0 as minimum and the result of the PHP function `mt_getrandmax` as the maximum.                              |
+| `getUniqueHash`   | Generates a unique hash string with an optional prefix.                                                                                                                              |
+
+## Examples
+
+### Generate a random string
+
+This example shows how to generate a random string of length 99, with the default character pool.
+
+```php
+    $this->mathRandom->getRandomString(99);
```

Comment:
Good PR , @drpayyne !

However, as a reader I would like to understand what is `$this->mathRandom` ?

Could you please add a small clarification before examples.

Thank you!

Reply:
Done, thank you!